internal/agent/device/spec/queue.go

```diff
@@ -130,6 +130,12 @@ func (m *queueManager) Next(ctx context.Context) (*v1alpha1.Device, bool) {
 	}
 
 	if exists {
+		if !requeue.downloadPolicySatisfied && !requeue.updatePolicySatisfied {
```

Comment:
Semi-related, but why is it safe to call `m.updatePolicy` above without considering the result of `exists`?